dedupe-infusion.js

```diff
@@ -0,0 +1,119 @@
+/*
+Dedupe Infusion
+
+Copyright 2015 Raising the Floor (International)
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/kettle/LICENSE.txt
+*/
+
+"use strict";
+
+var pkg = require("./package.json"),
+    path = require("path"),
+    fs = require("fs"),
+    glob = require("glob");
+    
+var dedupe = {};
+
+dedupe.shallowMerge = function (target/*,  ... */) {
+    for (var arg = 1; arg < arguments.length; ++ arg) {
+        var source = arguments[arg];
+        for (var key in source) {
+            target[key] = source[key];
+        }
+    }
+    return target;
+};
+
+// Utility to recursively delete a directory and its contents from http://www.geedew.com/2012/10/24/remove-a-directory-that-is-not-empty-in-nodejs/
+// Useful for cleaning up before and after test cases
+
+dedupe.deleteFolderRecursive = function (path) {
+    if (fs.existsSync(path)) {
+        fs.readdirSync(path).forEach(function (file) {
+            var curPath = path + "/" + file;
+            if (fs.lstatSync(curPath).isDirectory()) {
+                dedupe.deleteFolderRecursive(curPath);
+            } else { // delete file
+                fs.unlinkSync(curPath);
+            }
+        });
+        fs.rmdirSync(path);
+    }
+};
+
+dedupe.processFiles = function (infusions, node_modules) {
+    var infusionSegs = infusions.map(function (onePath) {
+        // Whilst the "path" module exports an entry "path.sep" purportedly holding this path separator,
+        // in practice this is not the one used by the grunt file expander
```

Comment:
Relevant but needs updating. "glob" uses just the same rules as "grunt" (which is not very surprising since it is based on the same underlying set of npm modules)